### PR TITLE
add channel list command

### PR DESCRIPTION
**Pull request by bobzoller**
Adds `agent-slack channel list` with Slack Web API support for:

- `users.conversations` by default (current user) or with `--user <user>`
- `conversations.list` with `--all`
- hard error for `--all` + `--user`
- always `exclude_archived=true` (follow-on could make this configurable)
- always `types=public_channel,private_channel,im,mpim` (follow-on could make this configurable)
- pagination via `--limit` and `--cursor`
- full conversation JSON output + optional `next_cursor`

Also includes tests and README/skill docs updates.

Validated a release build works for me locally, including pagination.

### CODE REVIEW

**Comment by AmethystLiang** [comment]
+1

File: .oxfmtrc.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "$schema": "./node_modules/oxfmt/configuration_schema.json",
+  "ignorePatterns": ["**/*.md", "**/*.json", "**/*.yaml", "**/*.yml"]
+}
```
Comment (nwparker) [inline .oxfmtrc.json]: Happy to consider changes here, but we can keep this out of this PR since I think it may warrant a discussion of it's own 🙏 !

**Comment by nwparker** [comment]
Thanks for the contribution @bobzoller !

Here is a review from Claude:
```
# PR #3 Review: Add channel list command

**Branch:** `bobzoller/channel-list` -> `master`
**+421 / -2** across 11 files

## Summary

Adds `agent-slack channel list` with support for listing conversations via `users.conversations` (default/per-user) or `conversations.list` (`--all`). Includes pagination, docs, and tests.

Overall this is a clean, well-structured PR that follows existing codebase patterns. A few items worth discussing:

---

## Issues

### 1. Unrelated file: `.oxfmtrc.json`

A new `.oxfmtrc.json` formatter config is included in this PR. It's unrelated to the channel list feature and should be split into its own commit or PR to keep the changeset focused.

### 2. Pagination model differs from `user list`

`user list` does **client-side aggregation** — it pages internally to fill the requested `--limit`, returning a single merged result. `channel list` does **single-page pass-through** — it sends `--limit` directly to the Slack API and returns one page, expecting the caller to paginate via `--cursor`.

Both approaches are valid, but the inconsistency could surprise consumers. Worth documenting the difference explicitly, or at minimum adding a note in the `channel list` help text that the caller must handle pagination with `--cursor`.

### 3. `normalizeLimit` could silently accept bad values from Slack's perspective

`normalizeLimit` clamps to `[1, 1000]`, but the Slack API `conversations.list` docs say the default is 100 and max is 1000. `users.conversations` has a default of 100. Passing `limit: 1` is technically valid but will result in very slow pagination. Consider whether a higher floor (e.g., 10) would be more practical, or at least document that very low limits are possible.

### 4. No unit tests for `normalizeLimit` edge cases

The `normalizeLimit` helper clamps values to `[1, 1000]` but has no direct tests for boundary conditions: `0`, `-1`, `1001`, `undefined`. The function is simple but given it's exported implicitly through the public API surface, edge-case coverage would be cheap to add.

### 5. `resolveUserId` exported as a side-effect of this PR

`src/slack/users.ts:79` — `resolveUserId` was changed from a private function to an `export`. This is needed for the channel command, but it widens the public API surface of the users module. This is fine, just calling it out as intentional.

---

## Nits

### 6. `ChannelListOptions` type is local to the command file

Other commands define their options types inline in the `args as [...]` cast (e.g., `user-command.ts`). Defining `ChannelListOptions` as a named type at the top of the file is slightly different from the pattern in `user-command.ts` but arguably clearer. Not a problem, just a style inconsistency.

### 7. `normalizeConversationsPage` is exported only for testing

`normalizeConversationsPage` in `src/slack/channels.ts:112` is exported and tested directly in `test/channels.test.ts`. This is fine for testability, but if the preference is to keep internal helpers unexported, it could be tested indirectly through `listUserConversations` / `listAllConversations` instead.

### 8. `DEFAULT_CONVERSATION_TYPES` constant is module-private

Good practice. Consider whether this should be a configurable option in the future (the PR description mentions this as a follow-on).

---

## What looks good

- **Follows existing patterns closely**: command registration, `withAutoRefresh`, `pruneEmpty`, error handling with `process.exitCode = 1` — all match the established conventions.
- **Clean `--all` / `--user` mutual exclusion**: validated early with a clear error message.
- **Thorough test coverage**: 7 test cases for the CLI command layer + 4 unit tests for the data layer. Covers happy paths, user resolution, error cases.
- **Documentation updated in all the right places**: README, SKILL.md, references/commands.md, references/output.md.
- **`resolveUserId` reuse**: Good call reusing the existing function rather than reimplementing user resolution.

---

## Verdict

**Approve with minor comments.** The core implementation is solid and well-tested. The main actionable items are (1) splitting out the `.oxfmtrc.json` change and (2) documenting the pagination model difference from `user list`.

```

Feedback 1-3 seem reasoanble.

LMK if you want to work on that otherwise I'm happy to also take this over, I think this is a great feature to add!

**Review by nwparker — COMMENTED** [review]
Thanks for the changes!

One last item, again happy to take over (always can let me know).
Namely, I think there's some token efficiency changes to be made.

# Change 1 - Channel List Less
We should reduce the fields returned on a `channel list`. There are alot right now. IMO we should just return: `id`, `name`, `is_private`, `is_archived`, `purpose.value`, `topic.value`, `num_members`
But what is users want more? See change 2 below !

# Change 2 - Channel Get
Add a `channel get` comamnd that DOES return more-or-less the all the fields (besides maybe some ultra useless one 🤷 ?). This way an agent could combine these methods in rarer cases where it needs more info.

# Change 3 - Channel Limit
_Apologies if I'm going a bit back-and-forth on this last one -- I think this my mistake_
We should let the user limit this down to 1 for token efficiency (if for whatever reason a user might want less channels).
I think Claude Code last time was just being dumb. My bad.

`src/slack/channels.ts`
```typescript
      132  }
      133
      134  function normalizeLimit(value: number | undefined): number {
      135 -  return Math.min(Math.max(value ?? 100, 10), 1000);
      135 +  return Math.min(Math.max(value ?? 100, 1), 1000);
      136  }
```

**Comment by nwparker** [comment]
Sorry about this! I had to rename the default branch from main to master for npx skills compatibility, and unfortunately GitHub auto-closed this PR and won't allow reopening it.

The branch and commits are all still intact — if you'd like to continue, you can open a new PR targeting master. Apologies for the inconvenience!

**Comment by nwparker** [comment]
I'll work on this one, I think this is a good feature to have

**Comment by nwparker** [comment]
Rebased follow-up is ready at #37: https://github.com/stablyai/agent-slack/pull/37

This ports the original channel list feature to current main, includes tests, and updates both README and the agent skill docs.

**Comment by bobzoller** [comment]
thanks for getting this over the line Neil, got distracted this week.
cheers!
